pkg/test/test.go

```diff
@@ -0,0 +1,38 @@
+// Package test implements the common structs used for defining
+// testcases.
+package test
+
+// Config struct is used to pass arbitrary information into the tests
+// such as IP address, expected FRU content, expected sensors
+type Config struct {
+
+	Host string;
+	Username string;
+	Password string;
+
+	// ... to be extended
+}
+
+// Result struct is used for returning test results from running a test
+type Result struct {
+
+	// what was tested
+	Name string
```

Comment:
Unnecessary redundancy since result should be part of the `Test` struct below.

Reply:
The name is required to have things like a tescase for temperature sensors, similar to how it's already done in the existing tool. Since we do not write a new struct and new function for every sensor, one test will test all the temp sensors and return multiple Result structs.

Each Result struct will then have the name of the sensor tested.
So when we print the results, we print the Name from struct Test and also the Name from struct Result.

Example:
```
PASS [Test] [redfish sensors] [sensor VR_TEMP_1] 
PASS [Test] [redfish sensors] [sensor VR_TEMP_2] 
PASS [Test] [redfish sensors] [sensor VR_TEMP_3] 
PASS [Test] [redfish sensors] [sensor SYS_AIR_INLET] 
```
or
```
FAIL [Test] [redfish sensors] [sensor VR_TEMP_1] Error: could not query redfish
PASS [Test] [redfish sensors] [sensor VR_TEMP_2] 
```